### DISCUSSION TITLE
Differences between magick, mogrify and convert? How about merge the tools?

### DISCUSSION THREAD

**Discussion by pstein** [ImageMagick/ImageMagick#7476] [General]
Its not really sure what the differences are between the tools "magick" mogrify" and "convert".

As far as I can with all 3 tools users can convert an image file from one format into another

Why not merging these tools into one?

**Comment by exekutive** [comment]
it is merged. You only need `magick`.  `Convert` is deprecated.
You can still use `magick mogrify` if you want to specify one file for input & output (overwrite)
Which version are you using?

Have you tried reading the documentation?
https://imagemagick.org/Usage/basics/
https://imagemagick.org/script/command-line-tools.php

**Comment by snibgo** [comment]
As exekutive says, the tools were merged, years ago, into a single program called `magick`.

The legacy tools (`convert`, `mogrify` etc) are provided so that old scripts will still work.

**Comment by 72757373656c6c** [comment]
I found this entry (#7476) and #7159 when searching for the same information. 

Bottom line, all **ImageMagick command-line utilities** are symlinked to **magick**. 

OS: Linux
imagemagick version: 7.1.1

```
2024-05-25 14:41 animate -> magick
2024-05-25 14:41 compare -> magick
2024-05-25 14:41 composite -> magick
2024-05-25 14:41 conjure -> magick
2024-05-25 14:41 convert -> magick
2024-05-25 14:41 display -> magick
2024-05-25 14:41 identify -> magick
2024-05-25 14:41 import -> magick
2024-05-25 14:41 magick
2024-05-25 14:41 magick-script -> magick
2024-05-25 14:41 mogrify -> magick
2024-05-25 14:41 montage -> magick
2024-05-25 14:41 stream -> magick
```

```
===> type -a magick;echo; type -a convert;echo;  stat -c '%N' /usr/bin/convert
magick is /usr/bin/magick
magick is /bin/magick

convert is /usr/bin/convert
convert is /bin/convert

'/usr/bin/convert' -> 'magick'
```
NOTE: On Linux, /bin is symlinked to /usr/bin, thus why `type -a` shows both.

I found the documentation, whether online or local (man pages), confusing --- Perhaps the documentation is in a transitional phase too. :thinking:  The bottom of the `magick` man page says **SEE ALSO** ImageMagick, which points to all the legacy tools. Some place there needs to be a huge banner with an explanation. There is only **magick** and here is the current documentation to match. :grin:

**Comment by 72757373656c6c** [comment]
I found several other discussions here on this topic, #7366 was the most helpful.

> Yes, use only magick in IM 7 when you want to do the equivalent of convert in IM 6. Preface all other IM 6 tools with magick in IM 7. So use magick identify, magick mogrify, magick montage, etc.
> https://github.com/ImageMagick/ImageMagick/discussions/7366#discussioncomment-9742379
